syntax = "proto3";
package proto;
import "google/protobuf/any.proto";


enum ChatType {
    UNKNOWN = 0;
    USER = 1;
    GROUP = 2;
    BOT = 3;
}

message Content {
    string text = 1;
}

message Msg {
    Header header = 1;
    google.protobuf.Any data = 2;

    message Header {
        string id = 1;
        string type = 2;
    }
}

message PushMessage {
    string message_id = 1;
    Sender sender = 2;
    string chatId = 4;
    ChatType chatType = 5;
    Content content = 6;
    int64 contentType = 7;
    int64 time = 8;

    message Sender {
        string id = 1;
        ChatType chatType = 2;
        string nickname = 3;
    }
}

message SendMessage{
    string message_id = 2; // 信息ID
    string chatId = 3; // 欲发送到的信息对象
    ChatType chatType = 4; // 欲发送到的信息对象的类别，1-用户，2-群聊，3-机器人
    Content content = 5;
    int64 contentType = 6; //1:文本, 2:图片, 3:markdown, 4:文件, 5:form(会灰屏), 6:文章, 7:表情, 8:HTML, 10:视频, 11:语音, 12:视频通话, 13:语音通话
}

message MessageSent {
    Header header = 1;
    int64 seq = 2;

    message Header {
        int64 code = 2;
        string msg = 3;
    }
}
